src/main/java/io/quarkus/gizmo/SignatureBuilder.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkus.gizmo;
+
+import io.quarkus.gizmo.Type.ClassType;
+import io.quarkus.gizmo.Type.ParameterizedType;
+import io.quarkus.gizmo.Type.TypeVariable;
+
+/**
+ * Builds a generic signature as defined in JVMS 17, chapter "4.7.9.1. Signatures".
+ * 
+ * @see SignatureElement#setSignature(String)
+ */
+public interface SignatureBuilder {
```

Comment:
It's minor but shouldn't this class be a class? It seems weird to me that we make it an interface given we don't really want implementations.

Reply:
We have implementations but we don't want users to provide their own implementations which is IMO very common... :shrug:.